### PR TITLE
[ONNX] Small refactor on RegisterInplaceNodeInBlocks

### PR DESCRIPTION
**Pull request by BowenBao**
Rename variables and remove unnecessary arguments.

### CODE REVIEW

File: torch/csrc/jit/passes/onnx/remove_inplace_ops_for_onnx.cpp [hunk 1]
```diff
@@ -490,52 +489,47 @@ void RegisterInplaceNodeInLoopBlocks(
...
 // Register inplace op node inputs/outputs through the blocks.
-void RegisterInplaceNodeInBlocks(
-    Value* orig_data,
-    Value* new_inplace_node,
-    Node* block_node,
-    Block* outer_block,
-    Node* next_node) {
-  auto cur_node = next_node;
+void RegisterInplaceNodeInBlocks(Value* orig_data, Value* new_data) {
+  Node* inplace_node = new_data->node();
```
Comment (jiafatom) [inline torch/csrc/jit/passes/onnx/remove_inplace_ops_for_onnx.cpp:500]: auto

File: torch/csrc/jit/passes/onnx/remove_inplace_ops_for_onnx.cpp [hunk 2]
```diff
@@ -439,29 +439,28 @@ void RegisterInplaceNodeInIfBlocks(
...
 //      -> (%27, %61)
-void RegisterInplaceNodeInLoopBlocks(
-    Value* orig_data,
-    Value* new_inplace_node,
-    Node* block_node,
-    Block* outer_block,
-    Node* next_node) {
-  if (next_node->kind() != prim::Loop)
+void RegisterInplaceNodeInLoopBlocks(Value* orig_data, Value* new_data) {
+  Node* inplace_node = new_data->node();
```
Comment (jiafatom) [inline torch/csrc/jit/passes/onnx/remove_inplace_ops_for_onnx.cpp:443]: auto

File: torch/csrc/jit/passes/onnx/remove_inplace_ops_for_onnx.cpp [hunk 3]
```diff
@@ -439,29 +439,28 @@ void RegisterInplaceNodeInIfBlocks(
...
+  Block* block = inplace_node->owningBlock();
+  Node* block_node = block->owningNode();
```
Comment (jiafatom) [inline torch/csrc/jit/passes/onnx/remove_inplace_ops_for_onnx.cpp:445]: Handle the case `block_node == nullptr` ?

**Review by jiafatom — APPROVED** [review]
Looks good to me.

**Comment by BowenBao** [comment]
Closing and merged with #51795

**Pull request closed without merging**